amethyst_renderer/src/pass/flat.rs

```diff
@@ -98,39 +92,40 @@ where
     fn apply<'a, 'b: 'a>(
         &'a mut self,
         supplier: Supplier<'a>,
-        (camera, mesh, material, global): (
+        (camera, mesh_storage, mesh, material, global): (
             Option<Fetch<'b, Camera>>,
-            ReadStorage<'b, M>,
-            ReadStorage<'b, N>,
+            Fetch<'a, AssetStorage<Mesh>>,
+            ReadStorage<'b, MeshHandle>,
+            ReadStorage<'b, Material>,
             ReadStorage<'b, T>,
         ),
-    ) -> DrawFlatApply<'a, V, M, N, T> {
+    ) -> DrawFlatApply<'a, V, T> {
         DrawFlatApply {
-            camera: camera,
-            mesh: mesh,
-            material: material,
-            global: global,
-            supplier: supplier,
+            camera,
+            mesh_storage,
+            mesh,
+            material,
+            global,
+            supplier,
             pd: PhantomData,
         }
     }
 }
 
-pub struct DrawFlatApply<'a, V, M: Component, N: Component, T: Component> {
+pub struct DrawFlatApply<'a, V, T: Component> {
     camera: Option<Fetch<'a, Camera>>,
-    mesh: ReadStorage<'a, M>,
-    material: ReadStorage<'a, N>,
+    mesh_storage: Fetch<'a, AssetStorage<Mesh>>,
```

Comment:
You need `AssetStorage<Texture>` as well

Reply:
Know, it's WIP

Reply:
Nearly done the passes though